domain/src/main/java/camp/nextstep/edu/domain/calculator/Expression.kt

```diff
@@ -53,8 +53,9 @@ class Expression(
         thunk.invoke(toString())
     }
 
-    fun get() =
-        expressions.toList().also { expressions.clear() }
+    fun clear() = expressions.clear()
+
+    fun get(): List<ExpressionItem> = expressions
```

Comment:
코틀린에서는 기본적으로 프로퍼티로 되어있습니다.
따라서 expressions도 getter를 가지고 있는데요~
이 경우 get함수를 만드는 것보다 아래처럼 사용하는 것도 좋습니다!
```suggestion
    private val _expressions  = mutableListOf<ExpressionItem>()
    val expressions: List<ExpressionItem>
        get() = expressions.toList()
```

또한 캡슐화를 위해 Backing property를 사용하는 경우, 혹은 get() 함수를 구현할 때 mutable한 리스트를 그대로 반환해주면 문제가 발생합니다.
이 경우 expressions는 List타입이기 때문에 불변타입으로 반환될 것이라고 생각이 되겠지만, mutable한 리스트의 레퍼런스가 그대로 넘어가기 때문에 레퍼런스를 이용하여 expressions를 외부 클래스에서 조작할 수가 있게 됩니다.
이런 경우 완전한 캡슐화가 이뤄지지 않기 때문에 `expressions.toList()` 와 같은 형식으로 반환해주는 것이 좋습니다.

참고: https://stackoverflow.com/questions/71046397/backing-property-vs-getter-for-returning-a-less-specific-type

Reply:
1. 프로퍼티에 접근해서 가져오게되면 expressions 라는 이름으로 가져와야하는데 맥락상 Expression.get() 이 좀 더 네이밍상 맞아보여서 따로 getter 를 정의했었어요
2. .toList 로 하지 않았던건 외부에서 그런식으로 쓸리가 없을것이다 라는 믿음기반이긴 한데... 바꾸는게 좋아보이네요 :)

Reply:
1번의 경우 저도 그랬을거라는 생각이 들더라구요..!
이 경우에는 value와 같이 expression의 값이라는 의미가 들어가는 변수 네이밍을 사용하셔도 좋습니다!